keps/sig-multicluster/4322-cluster-inventory/README.md

```diff
@@ -0,0 +1,1021 @@
+-->
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+We try to summarize the most common properties that should be able to
+represent a cluster and support the above use case at the minimum scope.
+
+The target consumers of the API are users who manage the clusters and
+other tools to understand the clusters concept from the API for
+multicluster scheduling, workload distribution and cluster management.
+The API aims to provide the information for the consumers to answer the
+below questions:
+
+* Is the cluster under management?
+* How can I select a cluster?
+* Is the cluster healthy?
+* Does the cluster have certain capabilities or properties?
+* Does the cluster have sufficient resources?
+
+### Cluster Name
+
+It is required that cluster name is unique for each cluster, and it
+should also be unique among different providers (cluster manager). There
+are two option to ensure it is uniqueness
+
+#### Option 1
+
+The `metadata.name` of the cluster should never be set upon creation.
+Only `metadata.generateName` can be set.
+
+#### Option 2
+
+The `metadata.name` and `metadata.generateName` must have prefix which
+should be the same as the `spec.clusterManager.name`. Different cluster
+manager must set a different value of `spec.clusterManager.name` when
+the cluster is created.
+
+These two options can be achieved by a validating webhook or a
+`ValidatingAdmissionPolicy` resource.
+
+### Spec
+
+#### Display name
+
+It is a human-readable name of the cluster set by the consumer of the
+cluster.
+
+#### Cluster Manager
+
+An immutable field set by a cluster manager when this cluster resources
+is created by it. Each cluster manager instance should set a different
+values to this field.
+
+In addition, a predefined label with key "x-k8s.io/cluster-manager" needs
+to be added by the cluster manager upon creation. The value of the label
+MUST be the same as the name of the cluster manager. The purpose of this
+label is to make filter clusters from different cluster managers eaiser.
+
+### Status
+
+#### Version
+
+Kubernetes version of the cluster
+
+Versions of the kubernetes can let consumers understand the capability
+of the kubernetes, such as what API is supported.
+
+With recent conversations about kube-apiserver and enabled featureset version,
+it is possible to incorporate other version relating to the cluster, such as
+minimum kubelet version, maximum kubelet version, and enabled featureset version.
+
+#### Resources
+
+Resources include capacity and allocatable, which is the sum of the
+capacity/allocatable of nodes on the cluster. It is useful for users to
+know the sizing of the cluster, and also schedulers to sort or bin pack
+the workload to clusters.
```

Comment:
I brought this up in the doc, but capacity/allocatable needs definition. What are we showing? CPUs and type? GPUs? there are many resources and they are not fungible. Sum of CPU is not a useful metric in a cluster with mixed infra. Even with only one CPU type, there are many cases where it breaks down: e.g. 100 CPU free, but only 8 CPU/node and I'm trying to schedule a 32 CPU pod

Reply:
removed the resource field, and added the resource base scheduling as the beta plan